drivers/gpu/drm/panel/panel-simple.c

```diff
@@ -4681,23 +4684,123 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "osddisplays,osd101t2045-53ts",
 		.data = &osd101t2045_53ts
+	}, {
+		/* Must be the last entry */
+		.compatible = "panel-dsi",
+		.data = &panel_dsi,
 	}, {
 		/* sentinel */
 	}
 };
 MODULE_DEVICE_TABLE(of, dsi_of_match);
 
+
+/* Checks for DSI panel definition in device-tree, analog to panel_dpi */
+static int panel_dsi_dt_probe(struct device *dev,
+			  struct panel_desc_dsi *desc_dsi)
+{
+	struct panel_desc *desc;
+	struct display_timing *timing;
+	const struct device_node *np;
+	const char *dsi_color_format;
+	const char *dsi_mode_flags;
+	int ret;
+
+	np = dev->of_node;
+
+	desc = devm_kzalloc(dev, sizeof(*desc), GFP_KERNEL);
+	if (!desc)
+		return -ENOMEM;
+
+	timing = devm_kzalloc(dev, sizeof(*timing), GFP_KERNEL);
+	if (!timing)
+		return -ENOMEM;
+
+	ret = of_get_display_timing(np, "panel-timing", timing);
+	if (ret < 0) {
+		dev_err(dev, "%pOF: no panel-timing node found for \"panel-dsi\" binding\n",
+			np);
+		return ret;
+	}
+
+	desc->timings = timing;
+	desc->num_timings = 1;
+
+	of_property_read_u32(np, "width-mm", &desc->size.width);
+	of_property_read_u32(np, "height-mm", &desc->size.height);
```

Comment:
Up to here in this function is all common with panel_dpi_probe.

If we change the 
```
if (desc == &panel_dpi) {
```
at https://github.com/torvalds/linux/blob/master/drivers/gpu/drm/panel/panel-simple.c#L593 to
```
if (desc == &panel_dpi || desc == &panel_dsi) {
```
don't you get that for free?

Reply:
The flow of DSI panels is quite seperate to all the other types. What you propose is what I had done previously but then decided against it as it breaks with how the driver is currently constructed (note that two drivers are actually registered, one for panel-simple and one for panel-simple-dsi).
I'm unsure tbh what the best approach is but yes there is a fair chunk of duplicate code but panel_dpi_probe assumes a dpi panel and writes dpi specific properties.
You could to some degree merge these two probe functions but it would require then two seperate spaces where we parse the device tree, one for the DSI port specific aspects set in panel_desc_dsi in panel_simple_dsi_probe and in panel_dpi probe with some if else required to set the dpi or dsi specific settings of panel_desc.

Separating the two despite the overlap in function and following the currently set flow of panel-simple-dsi seemed like the better approach.
Generally speaking panel-simple might require a broader restructuring as there are two competing ideas in there.
If I were to do that I would remove all of the existing panel definitions, move them into device tree overlays and have the driver be fully device tree defined but I don't think that would be acceptable as it would be a breaking change and given how contentious this was in the past would probably mean a lengthy discussion with upstream first.

Reply:
`panel_dpi_probe` currently takes `*dev` and a `struct panel_simple *panel`. It only uses `*panel` to fill in the `desc` field.

Modify it to take a `struct panel_desc **desc`, and pass in &panel-desc from both dpi and dsi paths. The DSI probe can then continue and read the DSI specific bits.

Yes it would leave a couple of minor and inconsequential DPI related things executed in the DSI path, but if viewed necessary those could be split out.

Reply:
Ok! So my approach so far was to only add and not change anything. The only reason for that was to appease to an upstreaming process as much as possible. If you think it would be fine to make some changes to the existing things then I would propose to keep the proping functions seperate but pull out the panel timing and panel-common property parsing into a new function that takes on the same role as the EDID parsing does. That function adds additional timings to a panel that were found on the EDID connected to the DT defined I2C port.
That feels like a more generic way and would allow any type of "connector" to get timings from device tree and the same wouldn't need to be done for say LVDS.
I'm a bit unsure about the dynamic creation of a panel as discussed above though. Could stick with having some dummy structs for now or see to handle this a bit more properly. A bit of a question of scope I guess, there is a lot that could technically be improved overall.
What are your thoughts?